bar.go

```diff
@@ -1,22 +1,94 @@
 package desktop
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os/exec"
+	"path/filepath"
+	"strings"
 
 	"fyne.io/fyne"
 	"fyne.io/fyne/layout"
-	"fyne.io/fyne/widget"
 )
 
-func newBar() fyne.CanvasObject {
-	return fyne.NewContainerWithLayout(layout.NewHBoxLayout(),
-		layout.NewSpacer(),
-		widget.NewButton("Browser", func() {
-			exec.Command("chromium").Start()
-		}),
-		widget.NewButton("Terminal", func() {
-			exec.Command("xterm").Start()
-		}),
-		layout.NewSpacer(),
-	)
+var appBar *Bar
+var fyconSize = 32
+var fyconTheme = "Papirus"
+var icons []*Fycon
+
+type barStackListener struct {
+}
+
+func barCreateFycon(taskbar bool, fdoDesktop *FdoDesktop, win Window) *Fycon {
```

Comment:
In Go we avoid duplication so "fdoDesktop *FdoDesktop" is not really best practice.
A descriptive but short name is preferred, such as "data", "source" or "metadata".